### ISSUE TITLE
current in this.input.current is misinterpreted as a prop

### ISSUE THREAD

**Issue by lagesag** [yannickcr/eslint-plugin-react#2101]
After update from 7.11.1 to 7.12.0 we started seeing this problem. Originally, the error occurred on a ref (e.g. `this.input.current`, where `this.input` was created in the constructor using `React.createRef()`), but here's a more abstract minimal example:

```javascript
import React from "react";

export class D extends React.Component {

    constructor(props) {
        super(props);
        this.x = {y: "z"};
    }

    componentDidUpdate() {
        // react/prop-types: 'y' is missing in props validation
        console.log(this.x.y);
    }

}
```

Oddly enough, only `componentDidUpdate()` seems to be affected:

```javascript
import React from "react";

export class D extends React.Component {

    constructor(props) {
        super(props);
        this.x = {y: "z"};
    }

    componentDidMount() {
        console.log(this.x.y); // NO error
    }

    componentDidUpdate() {
        console.log(this.x.y); // ERROR
    }

    render() {
        return this.x.y; // NO error
    }

}
```

(eslint 5.11.1)

**Comment by radoslawgrochowski** [comment]
``` 
  shouldComponentUpdate(nextProps, nextState) {
    return (
      nextState.disable !== this.state.disable ||
      nextState.fieldCount !== this.state.fieldCount
    );
  }
```

also got it @ (nextState.fieldCount and nextState.disable)

**Comment by ljharb** [comment]
Not sure if this is fixed by the unreleased #2099 or #2098

**Comment by dzek69** [comment]
I can confirm this.

Using `this.anything.anyName` inside `componentDidUpdate` causes `anyName` to be caught as prop without propType.

**Comment by ljharb** [comment]
Thanks for confirming.
